### ISSUE TITLE
Ways on different layers flagged because they cross but don't share a node

### ISSUE THREAD

**Issue by atiannicelli** [facebook/Rapid#1009]
### Description

When two highways with different layer tags cross but don't share a node Rapid is indicating that this is an issue. For many highways this is perfectly acceptable. Maybe the logic is expecting a bridge or tunnel??? in parking building with multiple layers people end up just adding tags to all the ways to indicate the layer.

### Version

2.0.3

### What browser are you seeing the problem on? What version are you running?

Chrome v115.0

### The OS you're using

mac

### Steps to reproduce

go here and look at the "issues"

https://rapideditor.org/rapid#background=NC_Latest_Ortho_Natural&datasets=fbRoads,msBuildings&disable_features=boundaries&id=w759921673&map=20.61/35.82127/-78.62064

### The browser URL at the time you encountered the bug

https://rapideditor.org/rapid#background=NC_Latest_Ortho_Natural&datasets=fbRoads,msBuildings&disable_features=boundaries&id=w758004645&map=20.44/35.82125/-78.62064

### The auto-detected useragent string for your browser (leave blank if you're manually filling this form out)

Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36

**Comment by Bonkles** [comment]
@atiannicelli there are a LOT of issues displayed in the UI at this location, so I'd like confirmation of your issue.  Let me know if this is the kind of thing you mean: 

![image](https://github.com/facebook/Rapid/assets/1887955/81be2925-0408-4e96-882c-4dec200667dc)

Two parking aisles:
-  one aisle at level 2, shown selected, way id `w758004645` 
- one at level 0, not selected, way id `w759921673`, perpendicular to the first but not connected

Confirm that you think this shouldn't be called out as a warning, as ways on different levels may or may not connect with one another. 

@bhousel what do you think about this one?

**Comment by bhousel** [comment]
I agree that ways with different `level` tag shouldn't be flagged for crossing without a connection 👍 
Sounds like a regression!

**Comment by Bonkles** [comment]
I just re-tested this on rapid v1, and it did the same thing there. I wonder how long this has been a regression for...

**Comment by bhousel** [comment]
This should be the case for probably all things right?  not just highways?

**Comment by Bonkles** [comment]
> This should be the case for probably all things right? not just highways?

Great question. I'm currently reading about levels, layers, bridges, tunnels, etc.  and thinking about this a bit more to see what cases I'm missing.

**Comment by bhousel** [comment]
yeah watch out
`layer` is for the road/bridge/tunnel arrangement
`level` is for what floor inside a building the feature is on.

I guess this issue is about `level` in a parking garage.  (Sorry, I was confused and read too quickly!)

**Comment by Bonkles** [comment]
Hopped on a quick call with @atiannicelli  to show him the new validation behavior- he approves.

**Comment by bhousel** [comment]
> Hopped on a quick call with @atiannicelli to show him the new validation behavior- he approves.

Awesome!  thanks for the quick fix!   🙌
